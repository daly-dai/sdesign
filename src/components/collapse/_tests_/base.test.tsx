import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import { describe, expect, it, vi } from 'vitest';

import SCollapse from '../index';

import mountTest from '@daly/sdesign/tests/mountTest';

describe('SCollapse', () => {
  mountTest(SCollapse);

  mountTest(() => <SCollapse />);

  it('should render correctly', () => {
    const { container } = render(<SCollapse />);
    expect(container).toMatchSnapshot();
  });

  it('should toggle collapse state on click', () => {
    const setCollapseMock = vi.fn();
    const onExpandMock = vi.fn();

    const { getByText } = render(
      <SCollapse
        collapse={false}
        setCollapse={setCollapseMock}
        onExpand={onExpandMock}
      />,
    );

    const button = getByText('收起');
    fireEvent.click(button);
    expect(setCollapseMock).toHaveBeenCalledWith(true);
    expect(onExpandMock).toHaveBeenCalledWith(true);
  });

  it('should toggle collapse state on button click', () => {
    const setCollapseMock = vi.fn();
    const onExpandMock = vi.fn();
    const { getByText } = render(
      <SCollapse
        collapse={false}
        setCollapse={setCollapseMock}
        onExpand={onExpandMock}
      />,
    );

    // 初始状态应该是“展开”和图标没有旋转
    expect(getByText('收起')).toBeInTheDocument();

    // 点击按钮
    fireEvent.click(getByText('收起'));

    // 验证 setCollapse 和 onExpand 被调用，并且状态变为“收起”
    expect(setCollapseMock).toHaveBeenCalledTimes(1);
    expect(setCollapseMock).toHaveBeenCalledWith(true);
    expect(onExpandMock).toHaveBeenCalledTimes(1);
    expect(onExpandMock).toHaveBeenCalledWith(true);
    expect(getByText('收起')).toBeInTheDocument();
  });
});
