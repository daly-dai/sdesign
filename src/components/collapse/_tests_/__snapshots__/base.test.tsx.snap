// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SCollapse > should render correctly 1`] = `
<div>
  <button
    class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-link"
    style="font-size: 14px; padding: 4px 0px;"
    type="button"
  >
    <span>
      收起
    </span>
    <span
      aria-label="caret-down"
      class="anticon anticon-caret-down"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="caret-down"
        fill="currentColor"
        focusable="false"
        height="1em"
        style="transform: rotate(180deg);"
        viewBox="0 0 1024 1024"
        width="1em"
      >
        <path
          d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`SCollapse should render 1`] = `
<div>
  <button
    class="ant-btn css-dev-only-do-not-override-1r287do ant-btn-link"
    style="font-size: 14px; padding: 4px 0px;"
    type="button"
  >
    <span>
      收起
    </span>
    <span
      aria-label="caret-down"
      class="anticon anticon-caret-down"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="caret-down"
        fill="currentColor"
        focusable="false"
        height="1em"
        style="transform: rotate(180deg);"
        viewBox="0 0 1024 1024"
        width="1em"
      >
        <path
          d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`SCollapse should render correctly 1`] = `
<div>
  <button
    class="ant-btn css-dev-only-do-not-override-1r287do ant-btn-link"
    style="font-size: 14px; padding: 4px 0px;"
    type="button"
  >
    <span>
      收起
    </span>
    <span
      aria-label="caret-down"
      class="anticon anticon-caret-down"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="caret-down"
        fill="currentColor"
        focusable="false"
        height="1em"
        style="transform: rotate(180deg);"
        viewBox="0 0 1024 1024"
        width="1em"
      >
        <path
          d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
        />
      </svg>
    </span>
  </button>
</div>
`;
